File: back-end/Students/Students.Data/Models/Student.cs
using System;
using System.Collections.Generic;

namespace Students.Data.Models
{
    public class Student
    {
        public Guid Id { get; set; }
        public string FirstName { get; set; }
        public string SecondName { get; set; }
        public Guid GroupId { get; set; }
        public int Semester { get; set; }
        public bool IsExpelled { get; set; }
        public bool IsDormitory { get; set; }
        public Group Group { get; set; }
        public List<Marks> StudentMarks { get; set; }
    }
}

//отчислен/нет?

//статус

File: back-end/students/src/main/java/com/src/students/model/Student.java
package com.src.students.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.OneToMany;
import javax.persistence.Table;
import java.util.List;


@JsonIgnoreProperties({"hibernateLazyInitializer"})
@Entity
@Table(name = "students")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Student {
    @Id
    @Column(name = "student_id")
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer id;
    @Column(name = "first_name")
    private String firstName;
    @Column(name = "second_name")
    private String secondName;
    @Column(name = "course")
    private Integer course;
    @Column(name = "specialty_id")
    private Integer specialtyId;
    @JsonProperty("isExpelled")
    @Column(name = "is_expelled")
    private boolean isExpelled = false;
    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "student_id")
    private List<Mark> marks;
}
